doc/userdoc/styleguide/styleguide.rst

```diff
@@ -0,0 +1,566 @@
+
+- Be consistent - Follow this style guide and relevant templates
+
+- Be specific - Avoid vague language; clarity is key
+
+- Focus your message - Start with what's most important, and keep text relevant
+  within given section
+
+- Write for your reader - Try to understand what your reader needs to know;
+  include context to what you're saying
+
+- Provide examples - Show don't tell; if you can use code snippets, screenshots,
+  or figures to illustrate a point, do so!
+
+- Avoid terms like simply, just, or easy - your knowledge does not always equal
+  your readers.  Don't make your reader feel bad because they -simply- didn't
+  understand something.
+
+- Write positively -  Use positive language rather than negative language
+
+Remember that the primary purpose of the documentation is to provide
+information to NEST users who are looking for it.
+
+Voice
+~~~~~~
+
+Voice is *how* we talk to people through writing. The voice encompasses
+substance, style, and tone.
+
+Our voice is
+
+- empathetic - we write thinking of who we are writing for and what their needs
+  are
+- relaxed - we write in natural, conversational way rather than an authoritative
+  way
+
+Use 'you' to indicate the reader and 'we' to indicate yourself (the writer and
+possibly the NEST team).
+
+The language we use
+~~~~~~~~~~~~~~~~~~~
+
+We follow spelling and grammar rules of American English.
+
+The markup language we use is reStructuredText. This includes documentation
+blocks in files written in Python and cpp.
+
+We avoid slang or jargon that is unnecessary or could be confusing to readers.
+
+We do, however, use terminology appropriate to NEST and neuroscience when
+specific and exact terms are needed, which is often. But explanations or
+references should be provided for clarity in introductory texts.
+
+If you have further questions about style not addressed here, see <<STANDARDIZED STYLE GUIDE>>
+
+Active voice
+~~~~~~~~~~~~
+
+Prefer the active voice, where the subject acts on object.
+
+If the object needs emphasis rather than subject, use passive voice. But ensure
+that you cannot improve the sentence by using the active voice.
+
+
+Types of formatting
+-------------------
+
+We use specific formatting marks to help readers scan through and find what
+they need, as well as provide references to important terms.
+
+
+Underlines for headings
+~~~~~~~~~~~~~~~~~~~~~~~
+
+ReStructuredText uses several types of underline markers for headings. It's
+important that the length of the underline is at least as long as the words
+in the heading.
+
+In general, we try to follow the pattern of
+
+* First heading: ===
+* Second heading: ---
+* Third heading: ~~~
+* Fourth heading: ^^^
+
+"Double quotes"
+~~~~~~~~~~~~~~~
+
+We use double quotes for strings in code, for example
+(nest.Create("iaf_psc_alpha")).
+
+Double or single quotes should not be used as an emphasis marker for text.
+
+
+\``Double backticks\``
+~~~~~~~~~~~~~~~~~~~~~~~
+
+Use double backticks for
```

Comment:
Right now there is no real logic; it's mostly based on what we already do in (most/some) cases. But I think this would be a good idea, as long as we consider what the final output will look like and ensure readability is not negatively affected.